pkg/cloud-provider/cloudapi/azure/azure_security.go

```diff
@@ -655,6 +660,85 @@ func (computeCfg *computeServiceConfig) getATGroupView(nepheControllerATSGNameTo
 	return enforcedSecurityCloudView, nil
 }
 
+// isValidAddressGroupSg finds if an ASG is in managed VNET, and it is a valid nephe created AddressGroup SG.
+func (computeCfg *computeServiceConfig) isValidAddressGroupSg(asgIDLowercase string,
```

Comment:
Can you merge isValidAddrGroup and isValidAppliedTo. Both share same logic.. Use bool to indicate if check if for AG or AT

Reply:
I purposely kept them separate as there is an extra operation in isValidAdressGroupSg() to figure out if asg is associated with network interface of managed vnet.  This is not needed in the other. Due to which function signatures are also different.
// proceed only if ASG is associated with network interfaces of managed VNET.
	vnetID, found := asgIDToVnetID[asgIDLowercase]
	if !found {
		return "", "", false
	}